### ISSUE TITLE
Google Sheet export via API

### ISSUE THREAD

**Issue by FrenchBen** [google/google-api-go-client#125]
I'm using the Google Drive v3 -
1. it's amazing that there are no APi calls for Google Sheets.
2. Because of #1I'm having to download the Sheet, to modify it and then re-upload it. 
3. I used the following call, after getting a Token for `drive.DriveScope`:

`resp, err := srv.Files.Export(fileID, "text/csv").Download()`
This results in the following error:

```
googleapi: got HTTP response code 500 with body: {
 "error": {
  "errors": [
   {
    "domain": "global",
    "reason": "internalError",
    "message": "Internal Error"
   }
  ],
  "code": 500,
  "message": "Internal Error"
 }
}
```

What am-I missing? is there a better way to work with Google Sheets?

**Comment by gmlewis** [comment]
There are no APIs for Go that I know of for working with Google Sheets.
There is, however, Google Apps Script, which is a JavaScript API to work with Sheets.
For more information, check out:
https://developers.google.com/apps-script/?hl=en

**Comment by FrenchBen** [comment]
So I have to do: Golang -> AppScript -> Golang ?

Any reason why such API do not exist? This seems to hint that there is some functionality:
https://developers.google.com/drive/v3/web/manage-downloads

**Comment by gmlewis** [comment]
That URL is just an alternative method of downloading the data via the REST API.
So you can't get the download to work... I'll see if I can duplicate the problem.

**Comment by FrenchBen** [comment]
thanks @gmlewis for the help!

**Comment by gmlewis** [comment]
It looks like a Drive problem... I was able to duplicate the error, and I get:
`Grid index required for text/csv`
so I tried changing the `mimeType` to `"text/tsv"` and it says `Conversion to text/tsv is not supported.`
I then tried `"text/ascii"` and that fails too.

So I finally dumped the `Export` call entirely and switched to `Get` and then get a "DownloadNotSupportedException".

I then switched back to the Drive v2 API.  This time, I get "The specified file does not support the requested alternate representation."

So I then tried:

```go
	url := fmt.Sprintf("https://www.googleapis.com/drive/v3/files/%s?alt=media", fileID)
	resp, err := client.Get(url)
	if err != nil {
		log.Fatalf("error opening %q: %v", url, err)
	}
	defer resp.Body.Close()
	buf, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		log.Fatal(err)
	}
	log.Printf("file:\n%s", buf)
```

and get a `bad request`.  So I switched back to "v2" and the error I get is: "The specified file does not support the requested alternate representation."

It looks to me like the Drive API is broken.  I don't know what to tell you at this point.  :-(
Maybe someone more familiar with Drive can comment.
